Student Application Dialog

29-29 opened the pull request:
Created a dialog to review the details of the application of a student before approving or rejecting.

lavahotsnacks committed:
removed commented out router-view

lavahotsnacks committed:
added onRowClick for "Show Details"

lavahotsnacks committed:
removed typo

lavahotsnacks committed:
fixed approve & reject buttons disappearing

29-29 committed:
changed v-slot to #

29-29 committed:
Removed filter in the top right and commented out filterOptions

Co-authored-by: Mangorangca <lavahotsnacks@users.noreply.github.com>

29-29 committed:
started rewriting the datatable

29-29 committed:
manually created the columns

29-29 committed:
added action buttons

29-29 committed:
Merge branch 'develop' of https://github.com/MSU-Students/student-advising-portal into good-datatable

29-29 committed:
added back approve and reject functionality

fencim committed:
Merge pull request #53 from MSU-Students/good-datatable

Rewriting Datatable

29-29 committed:
created view student application dialog

29-29 committed:
attached functionality to view application button

29-29 committed:
Merge branch 'develop' into good-student-app-dialog

fencim commented on src/components/admin/DataTable.vue:
```diff
@@ -1,60 +1,49 @@
 <template>
   <q-table
+    hide-bottom
```
adding this removes the pagination preventing from viewing the content on the next page

fencim commented on src/components/admin/ActionButtons.vue:
```diff
@@ -1,19 +1,79 @@
+  });
+}
+
+function approveApplication() {
+  Notify.create({
+    message: 'Are you sure?',
+    position: 'center',
+    actions: [
+      {
+        name: 'yes',
+        label: 'yes',
+        handler() {
+          TheWorkflows.emit({
+            type: 'approveApplication',
+            arg: {
+              payload: { ...props.props, data: { ...props.props.data } },
+            },
+          });
+        },
+      },
+      {
+        label: 'No',
+      },
+    ],
+  });
```
You may remove this, as this is temporary implementation. View Application is sufficient.

fencim commented on src/components/admin/ActionButtons.vue:
```diff
@@ -1,19 +1,79 @@
+      color="green"
+      class="q-mx-sm"
+      icon="check"
+      @click="approveApplication"
```
You may want to use this as to `viewApplication` which leads to `approveApplication`

fencim replied:
```suggestion
      @click="viewApplication"
```

fencim commented on src/components/admin/ActionButtons.vue:
```diff
@@ -1,19 +1,79 @@
-  </q-td>
+      <q-tooltip>Reject</q-tooltip>
+    </q-btn>
+    <q-btn round flat icon="more_horiz" @click="viewApplication">
+      <q-tooltip>View Application</q-tooltip>
+    </q-btn>
```
You can remove this button by keeping the approve button to view details

fencim commented on src/components/admin/DataTable.vue:
```diff
@@ -63,32 +52,37 @@
-  },
-] as { label: string; value: IRequest['status'] }[];
+
+// const filterOptions = [
+//   {
+//     label: 'Pending',
+//     value: 'pending',
+//   },
+//   {
+//     label: 'Approved',
+//     value: 'approved',
+//   },
+//   {
+//     label: 'Rejected',
+//     value: 'rejected',
+//   },
+// ] as { label: string; value: IRequest['status'] }[];
```
remove commented codes

29-29 committed:
applied suggested changes

- removed view application button
- removed hide-bottom attribute
- opened application rejection dialog on view application > reject button

29-29 committed:
removed commented out code

fencim reviewed (approved):
Good job
